### PR TITLE
[Objc] Update the name of the TV simulator to run the tests. Fixes #577

### LINKED ISSUE #577: [Objc] Tests fail in bots due to the name of the simulator.
On bot we are getting issues like the following:

13:31:42 	The requested device could not be found because no available devices matched the request.
13:31:42 
13:31:42 	Available destinations for the "Tests" scheme:
13:31:42 		{ platform:macOS, arch:x86_64 }
13:31:42 		{ platform:tvOS Simulator, id:2B9F6B92-402D-4120-8F78-6B51E6E5A3D7, OS:11.2, name:Apple TV }
13:31:42 		{ platform:tvOS Simulator, id:261D8A47-F65F-4FE6-98A3-2FA95CBCF25D, OS:11.2, name:Apple TV 4K }
13:31:42 		{ platform:tvOS Simulator, id:FE6FD93C-8129-412A-A3DB-38CDB357F036, OS:11.2, name:Apple TV 4K (at 1080p) }
13:31:42 
13:31:42 	Ineligible destinations for the "Tests" scheme:
13:31:42 		{ platform:tvOS, id:dvtdevice-DVTiOSDevicePlaceholder-appletvos:placeholder, name:Generic tvOS Device }
13:31:42 		{ platform:tvOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-appletvsimulator:placeholder, name:Generic tvOS Simulator Device }

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: objcgen/Make.config [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-MIN_MONO_URL=https://bosstoragemirror.blob.core.windows.net/wrench/mono-2017-02/9e/9ed0907eea13005d4e93b9c5f42c5c3015eba409/MonoFramework-MDK-5.0.0.18.macos10.xamarin.universal.pkg
+MIN_MONO_VERSION=5.8.0.118
+MAX_MONO_VERSION=5.8.0.118
+MIN_MONO_URL=https://bosstoragemirror.azureedge.net/wrench/mono-2017-10/73/7344771071aaf214d94ae1b8e4ba91c8961fe2d0/MonoFramework-MDK-5.8.0.118.macos10.xamarin.universal.pkg
 
 # Minimum Xamarin.iOS version
-MIN_XI_VERSION=10.11.0.136
-MAX_XI_VERSION=10.11.0.136
-MIN_XI_URL=https://bosstoragemirror.blob.core.windows.net/wrench/macios-mac-master/a3/a3264f64d6917ea8a1aab3884e4f51d229f31eda/xamarin.ios-10.11.0.136.pkg
+MIN_XI_VERSION=11.5.1.49
```
Comment (rolfbjarne) [inline objcgen/Make.config]: 11.6 is the current stable release, any reason not to use it?

File: objcgen/Make.config [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+MAX_XI_VERSION=11.5.1.49
+MIN_XI_URL=https://bosstoragemirror.azureedge.net/wrench/macios-mac-xcode9.2/11/113e518d9d04ce404414d810111af95a666d5280/xamarin.ios-11.5.1.49.pkg
 
 # Minimum Xamarin.Mac version
-MIN_XM_VERSION=3.5.0.144
-MAX_XM_VERSION=3.5.0.144
-MIN_XM_URL=https://bosstoragemirror.blob.core.windows.net/wrench/macios-mac-master/c3/c3cecd577f9d0813308df6d6d7159ed94073ddfb/xamarin.mac-3.5.0.144.pkg
+MIN_XM_VERSION=4.1.1.49
```
Comment (rolfbjarne) [inline objcgen/Make.config]: 4.0 is the current stable release.

File: objcgen/Make.config [hunk 3]
```diff
@@ -1,14 +1,14 @@
 # Minimum Mono version
-MIN_MONO_VERSION=5.0.0.18
-MAX_MONO_VERSION=5.0.99
-MIN_MONO_URL=https://bosstoragemirror.blob.core.windows.net/wrench/mono-2017-02/9e/9ed0907eea13005d4e93b9c5f42c5c3015eba409/MonoFramework-MDK-5.0.0.18.macos10.xamarin.universal.pkg
+MIN_MONO_VERSION=5.8.0.118
+MAX_MONO_VERSION=5.8.0.118
```
Comment (rolfbjarne) [inline objcgen/Make.config]: 5.4 is the current stable release.

File: objcgen/embedder.cs [hunk 4]
```diff
@@ -168,10 +168,13 @@ public int Generate (List<string> args)
 				directories.Add (Path.Combine (PlatformSdkDirectory, "Facades"));
 				foreach (var asm in Assemblies)
 					directories.Add (Path.GetDirectoryName (asm.Location));
-
+				directories.Add (Path.GetDirectoryName (GetPlatformAssembly ()));
+				
 				AssemblyName an = new AssemblyName (resolve_args.Name);
+				Console.WriteLine($"Assebly name is {an}");
```
Comment (rolfbjarne) [inline objcgen/embedder.cs]: Debug spew 😄 (and below too)

Comment (mandel-macaque) [reply objcgen/embedder.cs]: ups lol

**Comment by mandel-macaque** [comment]
Landing after a chat with @spouliot. Created issue #596 to track the mac failures.

**Pull request merged**